src/coins.cpp

```diff
@@ -160,33 +160,49 @@ bool CCoinsViewCache::GetNullifier(const uint256 &nullifier, ShieldedType type)
     return tmp;
 }
 
-void CCoinsViewCache::PushSproutAnchor(const ZCIncrementalMerkleTree &tree) {
+template<typename Tree, typename Cache, typename CacheIterator, typename CacheEntry>
+void CCoinsViewCache::AbstractPushAnchor(
+    const Tree &tree,
+    ShieldedType type,
+    Cache &cacheAnchors,
+    uint256 &hash
+)
+{
     uint256 newrt = tree.root();
 
-    auto currentRoot = GetBestAnchor(SPROUT);
+    auto currentRoot = GetBestAnchor(type);
 
     // We don't want to overwrite an anchor we already have.
     // This occurs when a block doesn't modify mapSproutAnchors at all,
     // because there are no joinsplits. We could get around this a
     // different way (make all blocks modify mapSproutAnchors somehow)
     // but this is simpler to reason about.
     if (currentRoot != newrt) {
-        auto insertRet = cacheSproutAnchors.insert(std::make_pair(newrt, CAnchorsSproutCacheEntry()));
-        CAnchorsSproutMap::iterator ret = insertRet.first;
+        auto insertRet = cacheSproutAnchors.insert(std::make_pair(newrt, CacheEntry()));
```

Comment:
Should be `cacheAnchors`.

Reply:
Caught in 501935ab34fa713b3e3ab283534bdfc1aae0b033 (which should be squashed into this commit once the PR is fully-reviewed).